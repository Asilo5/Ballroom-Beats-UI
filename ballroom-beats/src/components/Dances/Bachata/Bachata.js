import React, { Component } from 'react';
import {
    View,
    Text,
    Animated,
    Easing,
    TouchableOpacity,
    StyleSheet
} from 'react-native';

export default class Bachata extends Component {
    state = {
      pulses: [
        new Animated.Value(1),
        new Animated.Value(1),
        new Animated.Value(1),
        new Animated.Value(1)
      ],
      counters: [
          0,
          0,
          0,
          0
      ]
    };
   

   generateTiming = () => {
     return this.state.pulses.map((pulse) => {
       return [
          Animated.timing(pulse, {
              toValue: 3,
              duration: 439,
              easing: Easing.back()
          }),
          Animated.timing(pulse, {
              toValue: 1,
              duration: 439
          })
       ]
     }).flat();
   };

   countUp = (num) => {
     const subCounterList = [...this.state.counters];
     subCounterList[num]++;
     this.setState({ counters: subCounterList });
   };


   generateViews = () => {
     const colours = ["#F60091", "#F6811F", "#FFEB00", "#71C043"];

     return (
       <View style={styles.danceFloor}>
         <TouchableOpacity onPress={() => {this.countUp(0)}} >
             <Animated.View
                 style={{
                     transform: [
                         {
                             scaleX: this.state.pulses[0]
                         },
                         {
                            scaleY: this.state.pulses[0]
                        }
                     ],
                     margin: 20,
                     borderWidth: 10,
                     borderColor: "#F60091",
                     borderRadius: 10
                 }}  >
             </Animated.View>
         </TouchableOpacity>
         <TouchableOpacity onPress={() => {this.countUp(1)}} >
             <Animated.View
                 style={{
                     transform: [
                         {
                             scaleX: this.state.pulses[1]
                         },
                         {
                            scaleY: this.state.pulses[1]
                        }
                     ],
                     margin: 20,
                     borderWidth: 10,
                     borderColor: "#F6811F",
                     borderRadius: 10
                 }}  >
             </Animated.View>
         </TouchableOpacity>
         <TouchableOpacity onPress={() => {this.countUp(2)}} >
             <Animated.View
                 style={{
                     transform: [
                         {
                             scaleX: this.state.pulses[2]
                         },
                         {
                            scaleY: this.state.pulses[2]
                        }
                     ],
                     margin: 20,
                     borderWidth: 10,
                     borderColor: "#FFEB00",
                     borderRadius: 10
                 }}  >
             </Animated.View>
         </TouchableOpacity>
         <TouchableOpacity onPress={() => {this.countUp(3)}} >
             <Animated.View
                 style={{
                     transform: [
                         {
                             scaleX: this.state.pulses[3]
                         },
                         {
                            scaleY: this.state.pulses[3]
                        }
                     ],
                     margin: 20,
                     borderWidth: 10,
                     borderColor: "#71C043",
                     borderRadius: 10
                 }}  >
             </Animated.View>
         </TouchableOpacity>
       </View>
     )

   };

    render() {
        return (
          <View style={styles.bachataComponent}>
              <TouchableOpacity style={styles.start} onPress={() => this.props.start(this.generateTiming())}>
                  <Text style={styles.startText}>Start</Text>
              </TouchableOpacity>
              <View style={styles.stepsContainer}>
                  {this.generateViews()}
              </View>
          </View>
        );
    }
}

const styles = StyleSheet.create({
    danceFloor: {
      display: "flex",
      flexDirection: "column",
    },
    upperSteps:{
      display: 'flex',
      flexDirection: "row",
      width: '70%',
      height: 100,
      justifyContent: 'space-between'
    },
    upperTwoSteps: {
      display: 'flex',
      flexDirection: "row",
    },
    lowerSteps:{
      display: 'flex',
      flexDirection: "row",
      width: '70%',
      justifyContent: 'space-between'
    },
    lowerTwoSteps:{
      display: 'flex',
      flexDirection: "row",
    },
    stepsContainer: {
      padding: 10,
      left: 55,
      marginTop: 60,
      marginBottom: 60
    },
    numberView: {
      backgroundColor: "#FFF",
      margin: 10,
    },
    startText: {
      fontSize: 30,
      fontWeight: "bold",
      textAlign: "center",
      padding: 20
    },
    start: {
      backgroundColor: '#1CE61C',
      margin: 40,
      width: '80%',
      left: 100,
      borderRadius: 50,
    }
  });

