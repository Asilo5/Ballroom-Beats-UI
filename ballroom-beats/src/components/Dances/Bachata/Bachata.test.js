import React from 'react';
import renderer from 'react-test-renderer';
import Bachata from './Bachata';


describe('Bachata', () => {

   it('should match snapshot', () => {
    const bachata = renderer.create(<Bachata
      start={jest.fn()}
      song={jest.fn()}
      stopDance={jest.fn()}
      stopMusic={jest.fn()}
      />).toJSON();
    expect(bachata).toMatchSnapshot();
   });

});
