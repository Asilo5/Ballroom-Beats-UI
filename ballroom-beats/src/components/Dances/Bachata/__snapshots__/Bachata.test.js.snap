// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bachata should match snapshot 1`] = `
<View>
  <View
    style={
      Object {
        "left": 55,
        "marginBottom": 60,
        "marginTop": 60,
        "padding": 10,
      }
    }
  >
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "height": 400,
          "justifyContent": "space-around",
          "right": 50,
          "top": 100,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "borderColor": "#FCFDF9",
              "borderRadius": 15,
              "borderWidth": 15,
              "margin": 20,
              "transform": Array [
                Object {
                  "scaleX": 1,
                },
                Object {
                  "scaleY": 1,
                },
              ],
            }
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "borderColor": "#F60091",
              "borderRadius": 15,
              "borderWidth": 15,
              "margin": 20,
              "transform": Array [
                Object {
                  "scaleX": 1,
                },
                Object {
                  "scaleY": 1,
                },
              ],
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "bottom": 40,
            "display": "flex",
            "marginBottom": 100,
          }
        }
      >
        <View
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "borderColor": "#F6811F",
                  "borderRadius": 15,
                  "borderWidth": 15,
                  "margin": 20,
                  "transform": Array [
                    Object {
                      "scaleX": 1,
                    },
                    Object {
                      "scaleY": 1,
                    },
                  ],
                }
              }
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "borderColor": "#FFEB00",
                  "borderRadius": 15,
                  "borderWidth": 15,
                  "margin": 20,
                  "transform": Array [
                    Object {
                      "scaleX": 1,
                    },
                    Object {
                      "scaleY": 1,
                    },
                  ],
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "borderColor": "#71C043",
                  "borderRadius": 15,
                  "borderWidth": 15,
                  "margin": 20,
                  "transform": Array [
                    Object {
                      "scaleX": 1,
                    },
                    Object {
                      "scaleY": 1,
                    },
                  ],
                }
              }
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "borderColor": "#03ABF0",
                  "borderRadius": 15,
                  "borderWidth": 15,
                  "margin": 20,
                  "transform": Array [
                    Object {
                      "scaleX": 1,
                    },
                    Object {
                      "scaleY": 1,
                    },
                  ],
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "borderColor": "#274FA2",
              "borderRadius": 15,
              "borderWidth": 15,
              "margin": 20,
              "transform": Array [
                Object {
                  "scaleX": 1,
                },
                Object {
                  "scaleY": 1,
                },
              ],
            }
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "borderColor": "#6F2C8F",
              "borderRadius": 15,
              "borderWidth": 15,
              "margin": 20,
              "transform": Array [
                Object {
                  "scaleX": 1,
                },
                Object {
                  "scaleY": 1,
                },
              ],
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
